sbi/neural_nets/net_builders/vector_field_nets.py

```diff
@@ -0,0 +1,1186 @@
+        )
+
+        # unsqueeze to broadcast to sequence dimension
+        attn_scale = attn_scale.unsqueeze(1)
+        attn_shift = attn_shift.unsqueeze(1)
+        attn_gate = attn_gate.unsqueeze(1)
+        mlp_scale = mlp_scale.unsqueeze(1)
+        mlp_shift = mlp_shift.unsqueeze(1)
+        mlp_gate = mlp_gate.unsqueeze(1)
+
+        # self-attention with adaptive ln
+        x_norm = self.norm1(x)
+        x_norm = x_norm * (attn_scale + 1) + attn_shift
+
+        # self-attention
+        self_attn_out, _ = self.self_attn(x_norm, x_norm, x_norm)
+        x = x + attn_gate * self_attn_out
+
+        # cross-attention with adaptive ln
+        x_norm = self.norm2(x)
+        x_norm = x_norm * (mlp_scale + 1) + mlp_shift
+
+        # cross-attention with conditioning (no adaptive ln here)
+
+        # project conditioning to hidden dimension
+        cond_emb = cond_emb.unsqueeze(-2)
+        cond_emb = self.cond_proj(cond_emb)
+        cond_emb = cond_emb.squeeze(-2)
+        cross_attn_out, _ = self.cross_attn(query=x_norm, key=cond_emb, value=cond_emb)
+        x = x + cross_attn_out
+
+        # mlp with adaptive ln
+        x_norm = self.norm3(x)
+        x_norm = x_norm * (mlp_scale + 1) + mlp_shift
+
+        # mlp
+        mlp_out = self.mlp(x_norm)
+        x = x + mlp_gate * mlp_out
+
+        return x
+
+
+class VectorFieldTransformer(VectorFieldNet):
+    """Transformer for vector field estimation.
+
+    This class implements a DiT-style transformer architecture
+    for vector field estimation, using adaptive layer normalization
+    for conditioning.
+    """
+
+    def __init__(
+        self,
+        input_dim: int,
+        condition_dim: int,
+        hidden_features: int = 100,
+        num_layers: int = 5,
+        num_heads: int = 4,
+        mlp_ratio: int = 4,
+        time_emb_dim: int = 32,
+        time_emb_type: str = "sinusoidal",
+        sinusoidal_max_freq: float = 1000.0,
+        fourier_scale: float = 30.0,
+        activation: Callable = nn.GELU,
+        is_x_emb_seq: bool = False,
+        **kwargs,
+    ):
+        """Initialize dit-style transformer vector field network.
+
+        Args:
+            input_dim: Dimension of input data (theta).
+            condition_dim: Dimension of conditioning data (x).
+            hidden_features: Dimension of hidden features.
+            num_layers: Number of transformer layers.
+            num_heads: Number of attention heads.
+            mlp_ratio: Ratio for mlp hidden dimension.
+            time_emb_dim: Dimension of time embedding.
+            time_emb_type: Type of time embedding ('sinusoidal' or 'fourier').
+            sinusoidal_max_freq: Maximum frequency for sinusoidal embedding.
+            fourier_scale: Scale for fourier embedding.
+            activation: Activation function.
+            is_x_emb_seq: Whether x_emb is a sequence (if true, cross-attention is
+                used). NOTE: Effective usage is somewhat blocked by #1414
```

Comment:
Is this still blocked by 1414?

Reply:
More specifically by #1324 (that you cannt use variably shaped conditions. But can test for static ones